### PR TITLE
jsonpatcher: fix gob-encode of old ts events

### LINKED ISSUE #443: dispatcher broken for old-style event with time.Time
I've just tested https://github.com/textileio/go-threads/pull/439 and got the gob encode error in the dispatcher for old-format event [here](https://github.com/textileio/go-threads/blob/master/db/dispatcher.go#L80)
```gob: type not registered for interface: map[string]interface {}```

So looks like an empty time.Time decoded into map[string]interface which is not auto-registred in the gob

### PR DESCRIPTION
**Pull request by requilence**
fixes #443

### CODE REVIEW

**Comment by requilence** [comment]
I've added gob.Encode checks into the existing test in `jsonpatcher_test.go` instead of separate test in the `db` pkg. This seems not very correct from, because this way we also need to do gob.Register in test so we don't actually test the `db` pkg. Let me know if you are ok with it, or it's better to do a separate test

**Review by sanderpick — APPROVED** [review]
Sweet, thanks!

**Comment by sanderpick** [comment]
I'm okay with this test... thanks 🙏

File: jsonpatcher/jsonpatcher_test.go [hunk 1]
```diff
@@ -35,18 +44,37 @@ func TestJsonPatcher_Migration(t *testing.T) {
...
+
 	err = cbornode.DecodeInto(node2.RawData(), &out)
 	if err != nil {
 		t.Errorf("failed to decode event with int64 timestamp: %v", err)
 	}
 	tsout := out.time()
+
+	b.Reset()
+	e = gob.NewEncoder(&b)
+	if err := e.Encode(tsout); err != nil {
```
Comment (sanderpick) [inline jsonpatcher/jsonpatcher_test.go:74]: Did you mean for this to be `e.Encode(out)` as above?

Comment (requilence) [reply jsonpatcher/jsonpatcher_test.go]: yeap, fixed

**Pull request merged**